internal/stats/stats_service.go

```diff
@@ -424,6 +439,43 @@ func (g gaugeOS) updateDescr(bt *binTextStats) {
 	bt.stats = append(bt.stats, makeTextStat(&g.openDecrCount, nfd))
 }
 
+type gaugeNearCache struct {
+	serviceName string
+	f           func() func(service string) NearCacheStatsGetter
```

Comment:
Why don't we directly store `func(service string) NearCacheStatsGetter` here?

Reply:
`f` is called before `hazelcast.Client` is initialized and has a chance to set the correct `NearCacheStatsGetter`.

Reply:
So if you move `c.ic.StatsService.SetNCStatsGetter` call into the if block (`if config.StatsEnabled {` in internal/client/client.go) we can simplify here as well. We can directly pass the function to the gaugeNearCache.